### PR TITLE
Register sdp lower precision autocast

### PR DESCRIPTION
**Pull request by lingzhi98**
Details can be seen in [Issue 7177](https://github.com/pytorch/xla/issues/7177).

### CODE REVIEW

**Review by qihqi — APPROVED** [review]

**Comment by JackCaoG** [comment]
hmm if you rebase the issue should be gone

**Comment by lingzhi98** [comment]
I think this PR need re-approve to run the tests.

**Comment by lingzhi98** [comment]
I can not reproduce the test_train_mp_mnist_amp.py failure in my machine. And it is weird that the training loss has decreased to 0.00066 which is same as I see in my machine but the accuracy is 0. I really don’t know why sdp affects the mnist. It’s obvious that this op is not used at all.

**Comment by JackCaoG** [comment]
Let me retrigger, it might be using the fake data so accuracy doesn't matter. It seems like CI just exited with an error code but I don't know why.

**Comment by lingzhi98** [comment]
The CI has passed, this PR can be merged.
